torch/optim/lr_scheduler.py

```diff
@@ -24,6 +25,19 @@ def __init__(self, optimizer, last_epoch=-1):
                                    "in param_groups[{}] when resuming an optimizer".format(i))
         self.base_lrs = list(map(lambda group: group['initial_lr'], optimizer.param_groups))
         self.last_epoch = last_epoch
+
+        # Following https://github.com/pytorch/pytorch/issues/20124
+        # We would like to ensure that `lr_scheduler.step()` is called after
+        # `optimizer.step()`
+        def with_counter(func):
+            @wraps(func)
+            def wrapper(*args, **kwargs):
+                wrapper.called += 1
+                return func(*args, **kwargs)
+            wrapper.called = 0
+            return wrapper
+
+        self.optimizer.step = with_counter(self.optimizer.step)
```

Comment:
I haven't really understood the approach that this PR is taking, but monkeypatching optimizer.step give me the jeebies. (I don't know if there is a way to do this otherwise, though.)

Reply:
OK, I see what's going on. I think I agree that this is the only way to do it.

Reply:
Yeah, it is quite a hack but I also don't know any better alternative.